### ISSUE TITLE
photospline code quality improvements (Trac #1183)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1177]
2 fixmes found in code:

private/cfitter/nnls.c:				/* FIXME: re-use AtA_F from modify call? */
resources/scripts/kyrre.py:	# XXX: FIXME: need to multiply in normalization factors for each tensor-product spline

3 static analysis issues: http://software.icecube.wisc.edu/static_analysis/2015-08-18-030301-2272-1/

code coverage: coverage is good for bspline.c and convolve.cbut really bad for cfitter, would it be possible to write a unit test for these which wouldn't take for ever?

Migrated from https://code.icecube.wisc.edu/ticket/1183
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:57", 
    "description": "2 fixmes found in code:\n\nprivate/cfitter/nnls.c:\t\t\t\t/* FIXME: re-use AtA_F from modify call? */\nresources/scripts/kyrre.py:\t# XXX: FIXME: need to multiply in normalization factors for each tensor-product spline\n\n3 static analysis issues: http://software.icecube.wisc.edu/static_analysis/2015-08-18-030301-2272-1/\n\ncode coverage: coverage is good for bspline.c and convolve.cbut really bad for cfitter, would it be possible to write a unit test for these which wouldn't take for ever?", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067117911749", 
    "component": "combo reconstruction", 
    "summary": "photospline code quality improvements", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-08-19T12:26:00", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160215T10:32:39: jvansanten commented:

> Static analysis fixed in r141601

**Comment by icecube-trac** [comment]
Trac update at 20160215T10:33:19: jvansanten commented:

> In changeset 141846:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="141846"
> Remove FIXMEs (see #1177)
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160215T13:15:39: 
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 141856:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="141856"
> Add a regression test for cfitter. Closes #1177.
> ```


* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:57: 
* olivas changed milestone from "IceRec V5" to ""
* olivas commented:

> Milestone IceRec V5 deleted
